master/docs/manual/worker-transition.rst

```diff
@@ -0,0 +1,1091 @@
+     - :py:meth:`buildbot.protocols.base.Connection.remoteGetWorkerInfo`
+
+
+   * - :py:class:`buildbot.protocols.pb.Connection`
+       constructor positional argument ``buildslave`` was renamed
+     - ``worker``
+
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.findBuildslaveId`
+       (introduced in nine)
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.findWorkerId`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.deconfigureAllBuidslavesForMaster`
+       (introduced in nine, note typo ``Buidslaves``)
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.deconfigureAllWorkersForMaster`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveConfigured`
+       (introduced in nine)
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.workerConfigured`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveConfigured`
+       method argument ``buildslaveid`` was renamed
+       (introduced in nine)
+     - ``workerid``
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.getBuildslave`
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.getWorker`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.getBuildslaves`
+       method argument ``_buildslaveid`` was renamed
+       (introduced in nine)
+     - ``_workerid``
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveConnected`
+       (introduced in nine)
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.workerConnected`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveConnected`
+       method argument ``slaveinfo`` was renamed
+       (introduced in nine)
+     - ``workerinfo``
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveConnected`
+       method argument ``buildslaveid`` was renamed
+       (introduced in nine)
+     - ``workerid``
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveDisconnected`
+       (introduced in nine)
+     - :py:meth:`buildbot.db.worker.WorkersConnectorComponent.workerDisconnected`
+
+   * - :py:meth:`buildbot.db.buildslave.BuildslavesConnectorComponent.buildslaveDisconnected`
+       method argument ``buildslaveid`` was renamed
+       (introduced in nine)
+     - ``workerid``
+
+
+   * - :py:meth:`buildbot.db.builds.BuildsConnectorComponent.getBuilds`
+       method argument ``buildslaveid`` was renamed
+       (introduced in nine)
+     - ``workerid``
+
+
+   * - :py:meth:`buildbot.db.builds.BuildsConnectorComponent.addBuild`
+       method argument ``buildslaveid`` was renamed
+       (introduced in nine)
+     - ``workerid``
+
+
+   * - :py:class:`buildbot.reporters.message.MessageFormatter`
+       template variable ``slavename``
+     - ``workername``
+
+Data API
+--------
+
+Python API changes:
```

Comment:
Yes, no backward compatibility for this (and for everything else that doesn't have explicit note about fallback support).